# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1385327180 -3600
#      Sun Nov 24 22:06:20 2013 +0100
# Node ID 8dd4275381474c6d9e354f592528605cc0eec935
# Parent  260b44c0ed690b8d8fe678e3b62124917228fd32
fix restoring the geometry of docked widgets (bug #40691)

* main-window.cc (set_window_layout): hide main win during restoring geometry

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -904,21 +904,25 @@ main_window::set_window_layout (QSetting
           else
             {
               widget->make_widget ();
               widget->setVisible (visible);       // not floating -> show
             }
         }
     }
 
-  // show main first but minimized to avoid flickering
+  // show main first but minimized to avoid flickering,
   // otherwise the name of a floating widget is shown in a global menu bar
   showMinimized ();
+  // hide again, otherwise the geometry is not exactly restored
+  hide ();
+  // restore geomoetry of main window
   restoreState (settings->value ("MainWindow/windowState").toByteArray ());
   restoreGeometry (settings->value ("MainWindow/geometry").toByteArray ());
+  // show main window
   show ();
 
   // show floating widgets after main win to ensure "Octave" in central menu
   foreach (octave_dock_widget *widget, float_and_visible)
     {
 #if ! defined (Q_OS_WIN32)
       widget->make_window ();
 #endif
